main.py

```diff
@@ -1,53 +1,83 @@
-        response = requests.request("GET", url, params=querystring)
-        # TODO: we don't really use print in programs. Please use logger (library logging)
-        # this should be logged to info channel
-        print(response.status_code)
-        # TODO: what happens if this response is not a correct one, like 404 or 500?
-        # You need some kind of error detection here: response.raise_for_status()
-        weather_data = response.json()
-        data.append(weather_data)
+    # read data from 'locations_list.json' file
+    with open(locations_list, 'r') as f:
+        file_content = f.read()
+        places = json.loads(file_content)
+
+    # iterate through the locations list
+        for place in places:
+            querystring = {"q": place["name"], "units": "metric", "appid": api_key}
+
+            # send a get request
+            try:
+                response = requests.request("GET", url, params=querystring)
+                response.raise_for_status()
+            except requests.exceptions.RequestException as error:
+                logger.error(error)
+                print(error)
+            else:
+                logger.info(response.status_code)
+                print(response.status_code)
+
+            weather_data = response.json()
+            data.append(weather_data)
 
     # Transform data
     formatted_data = []
+    json_formatted_data = []
     for d in data:
         place = d["name"]
         temp = d["main"]["temp"]
         humidity = d["main"]["humidity"]
         wind_speed = d["wind"]["speed"]
         description = d["weather"][0]["description"]
-        formatted_data.append(f"The weather conditions for {place}: "
-                              f" the temperature is {temp}C, "
-                              f"humidity: {humidity}%, "
-                              f"wind: {wind_speed}m/s, "
-                              f" {description}")
+        formatted_data.append([["name", place],
+                               ["temp", str(temp)],
+                               ["humidity", str(humidity) + '%'],
+                               ["wind", str(wind_speed)],
+                               ["description", description]
+                               ])
+        json_formatted_data.append({"name": place, "temp": str(temp),
+                                    "humidity": str(humidity) + '%',
+                                    "wind":str(wind_speed),
+                                    "description": description
+                                    })
 
     # Write data to file
-    # TODO: if we want to import it into table, this should be a machine readable format
-    # json is enough. but you could also try to save it as parquet file format as an excercise
-    with open("weather.txt", "w") as f:
-        f.write("\n".join(formatted_data))
+    with open("weather.json", "w") as f:
+        json.dump(json_formatted_data, f)
```

Comment:
w pracy inżyniera danych bardzo rzadko zapisuje się dane w formacie pojedynczego Jsona. Zazwyczaj przyjętym formatem jest JSON Line (https://jsonlines.org/) czyli po jednym JSON w każdej linijce.
Najłatwiej to zapisać takim kodem:
```
for record in json_formatted_data:
    print(json.dumps(record), file=output_file)
```
czyli iterujemy po rekordach w danych, i dla każdego rekordu zapisujemy jego formę json jako pojedyncza linia w pliku wynikowym. To najoszczędniejszy sposób zapisu do pliku.

Reply:
Rezultat tego wygląda tak:
{"name": "....", "...": "......"}
{"name": "....", "...": "......"}
{"name": "....", "...": "......"}
co pycham oznacza jako nieprawidłowy format. 
Ale rozumiem, że tak ma być